### PR TITLE
feat: Jupyterhub oauth support

### PR DESCRIPTION
**Pull request by omrishiv**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Adding an example of using custom oauth with jupyterhub. This is tested with Keycloak, but should work with Okta or any oauth provider. 

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ ] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Comment by omrishiv** [comment]
Addresses #576

File: ai-ml/jupyterhub/variables.tf [hunk 1]
```diff
@@ -55,6 +55,27 @@ variable "acm_certificate_domain" {
 }
 variable "jupyterhub_domain" {
   type        = string
-  description = "Enter sub-domain name for jupyterhub to be hosted,  e.g. eks.example.com. Only needed if auth mechanism is set to cognito"
+  description = "Enter sub-domain name for jupyterhub to be hosted,  e.g. eks.example.com. Only needed if auth mechanism is set to cognito or oauth"
+  default     = ""
+}
+variable "oauth_domain" {
```
Comment (vara-bonthu) [inline ai-ml/jupyterhub/variables.tf:61]: nit: Add a new line after each variable declaration for consistency. Same for all the new variables mentioned below

**Review by vara-bonthu — COMMENTED** [review]
Great work on this @omrishiv! Adding a new section in the Website document to enable `OAuth` is beneficial for this PR.

Please add a section titled `Type3 (Optional): Deployment with OAuth`. In this section, include a paragraph explaining how to enable OAuth and include screenshots similar to the ones for Cognito, if available.

**Comment by omrishiv** [comment]
@vara-bonthu thank you for the comments! Hopefully the commit addresses them

**Review by vara-bonthu — APPROVED** [review]
lgtm

**Pull request merged**